lms/djangoapps/courseware/grades.py

```diff
@@ -842,7 +842,9 @@ def _calculate_score_for_modules(user_id, course, modules):
         if module.category not in ignore_categories and (module.graded or module.has_score):
             module_score = scores_client.get(locations[index])
             if module_score:
-                module_percentages.append(module_score.correct / module_score.total)
+                correct = module_score.correct or 0
+                total = module_score.total or 1
+                module_percentages.append(correct / total)
```

Comment:
@douglashall. I have no prior experience with the module_score so I am not sure if module_score.correct and .total are float types or integer types. In case they are both integer types, then this is going to be integer-division which might lead to issues.

e.g. if total is 2 and correct is 1:

1/2 is going to result in 0 and not 0.5

The workaround would be module_percentages.append(float(correct) / total)